lib/instance-validator.js

```diff
@@ -320,17 +320,20 @@ class InstanceValidator {
    */
   _validateSchema(rawAttribute, field, value) {
     if (rawAttribute.allowNull === false && (value === null || value === undefined)) {
-      const validators = this.modelInstance.validators[field];
-      const errMsg = _.get(validators, 'notNull.msg', `${this.modelInstance.constructor.name}.${field} cannot be null`);
-
-      this.errors.push(new sequelizeError.ValidationErrorItem(
-        errMsg,
-        'notNull Violation', // sequelizeError.ValidationErrorItem.Origins.CORE,
-        field,
-        value,
-        this.modelInstance,
-        'is_null'
-      ));
+      const association = _.values(rawAttribute.Model.associations).find(association => association.foreignKey === rawAttribute.fieldName);
```

Comment:
You will need to filter/check if association is of `belongsTo` type

Reply:
I think it's a given at this point in the call stack, right?

- This is called with each attr from the `modelInstance`'s attributes list, so we know it's a defined attribute (e.g. an attr or a foreign key)
- It's wrapped into the association query: If this wasn't a belongs-to, we won't find a related association with a matching foreign key (given another model's foreign key is a `targetKey`)
- This is a "private" function, so we shouldn't need to worry about adverse calls

I _think_ my logic is correct here, but I don't mind either admitting I've missed something or adding some bolt-tightening logic. I figured I'd suggest saving a cycle if it's pertinent.

Reply:
You can save filter cycle and check for belongsTo directly in `.find`, but I think `BelongsTo` check is important here. It should be as simple as `instanceof BelongsTo` check